src/elements/fields/DateSelectorField/index.tsx

```diff
@@ -138,12 +140,20 @@ function DateSelectorField({
     return !disabledDates.includes(`${date.getMonth() + 1}-${date.getDate()}`);
   };
 
+  // Called when the input is changed
   const onDateChange = (newDate: any) => {
     newDate = newDate ?? '';
     setInternalDate(newDate);
     onChange(formatDateString(newDate, servarMeta));
   };
 
+  // Called when the date is selected, or field is blurred
+  const onDateComplete = (newDate: any) => {
```

Comment:
can we unify this function with onDateChange? the only difference is which callback funciton is called